packages/core/src/rum/DdRum.ts

```diff
@@ -0,0 +1,319 @@
+                    'DDdRum.stopAction called with the old signature'
+                );
+                const { type, name } = this.lastActionData;
+                return bufferVoidNativeCall(() =>
+                    this.nativeRum.stopAction(
+                        type,
+                        name,
+                        args[0] || {},
+                        args[1] || timeProvider.now()
+                    )
+                );
+            }
+            InternalLog.log(
+                'DdRum.startAction needs to be called before DdRum.stopAction',
+                SdkVerbosity.WARN
+            );
+        } else {
+            InternalLog.log(
+                'DdRum.stopAction was called with wrong arguments',
+                SdkVerbosity.WARN
+            );
+        }
+
+        return new Promise<void>(resolve => resolve());
+    };
+
+    addAction(
+        type: RumActionType,
+        name: string,
+        context: object = {},
+        timestampMs: number = timeProvider.now()
+    ): Promise<void> {
+        InternalLog.log(
+            `Adding RUM Action “${name}” (${type})`,
+            SdkVerbosity.DEBUG
+        );
+        return bufferVoidNativeCall(() =>
+            this.nativeRum.addAction(type, name, context, timestampMs)
+        );
+    }
+
+    startResource(
+        key: string,
+        method: string,
+        url: string,
+        context: object = {},
+        timestampMs: number = timeProvider.now()
+    ): Promise<void> {
+        InternalLog.log(
+            `Starting RUM Resource #${key} ${method}: ${url}`,
+            SdkVerbosity.DEBUG
+        );
+        return bufferVoidNativeCall(() =>
+            this.nativeRum.startResource(key, method, url, context, timestampMs)
+        );
+    }
+
+    stopResource(
+        key: string,
+        statusCode: number,
+        kind: ResourceKind,
+        size: number = -1,
+        context: object = {},
+        timestampMs: number = timeProvider.now(),
+        resourceContext?: XMLHttpRequest
+    ): Promise<void> {
+        const mappedEvent = this.resourceEventMapper.applyEventMapper({
+            key,
+            statusCode,
+            kind,
+            size,
+            context,
+            timestampMs,
+            resourceContext
+        });
+        if (!mappedEvent) {
+            /**
+             * To drop the resource we call `stopResource` and pass the `_dd.drop_resource` attribute in the context.
+             * It will be picked up by the resource mappers we implement on the native side that will drop the resource.
+             * This ensures we don't have any "started" resource left in memory on the native side.
+             */
+            bufferVoidNativeCall(() =>
+                this.nativeRum.stopResource(
+                    key,
+                    statusCode,
+                    kind,
+                    size,
+                    {
+                        '_dd.resource.drop_resource': true
```

Comment:
Should this be a constant somewhere instead? If it's only used here, I'm actually partial to just leaving it and not putting the definition farther away.

Reply:
So far we don't abstract constants on the JS side, as they're used at only 1 place.
I believe the best would be to have only 1 file used to generate iOS/Android/JS constants so we are sure they all match, but that's probably a bit overengineered for now